src/PHPUnit/Constraint/Crawler.php

```diff
@@ -1,16 +1,26 @@
 <?php
+
+declare(strict_types=1);
+
 namespace Codeception\PHPUnit\Constraint;
 
 use Codeception\Exception\ElementNotFound;
 use Codeception\Lib\Console\Message;
-use Symfony\Component\DomCrawler\Crawler as DomCrawler;
+use DOMElement;
+use Facebook\WebDriver\WebDriverBy;
+use PHPUnit\Framework\ExpectationFailedException;
 use SebastianBergmann\Comparator\ComparisonFailure;
+use Symfony\Component\DomCrawler\Crawler as SymfonyDomCrawler;
+use function strpos;
 
 class Crawler extends Page
 {
-    protected function matches($nodes) : bool
+    /**
+     * @param SymfonyDomCrawler $nodes
+     * @return bool
+     */
+    protected function matches($nodes): bool
```

Comment:
was there an issue that prevented setting type `SymfonyDomCrawler` in method signature?

Reply:
the base class [`vendor/phpunit/phpunit/src/Framework/Constraint/Constraint.php`](https://github.com/sebastianbergmann/phpunit/blob/master/src/Framework/Constraint/Constraint.php) does not have these parameters typed.
I don't feel comfortable overwriting that signature even if only a DomCrawler is used (or a `string` in the case of the methods you mentioned above).

What I did was add/update the docblock with the type that you suggested.

Reply:
Making this change also generates errors in the tests:

```shell
2) PhpBrowserTest: Submit form with queries
 Test  tests/unit/Codeception/Module/PhpBrowserTest.php:testSubmitFormWithQueries
                                                                                                                                                                                                                                                                                                 
  [PHPUnit\Framework\Exception] Declaration of Codeception\PHPUnit\Constraint\Crawler::matches(Symfony\Component\DomCrawler\Crawler $nodes): bool should be compatible with Codeception\PHPUnit\Constraint\Page::matches(string $other): bool at ../../../src/PHPUnit/Constraint/Crawler.php:18  
```